test/90_1_prm/test_appendix_g_prm.rb

```diff
@@ -396,6 +396,214 @@ def check_lpd(prototypes_base)
+          assert((thermal_zone.sizingZone.zoneHeatingDesignSupplyAirTemperature - setpoint_c).abs < 0.001, "heating design supply air temperature for #{thermal_zone.name} in the #{building_type} #{template}, #{climate_zone} model is incorrect. For zones with unit heaters, heating design supply air temperature should be #{setpoint_c} (90.1 Appendix G3.1.2.8.2)")
+        else
+          case thermal_zone.sizingZone.zoneHeatingDesignSupplyAirTemperatureInputMethod
+          when 'SupplyAirTemperatureDifference'
+            assert((thermal_zone.sizingZone.zoneHeatingDesignSupplyAirTemperatureDifference - delta_t_r).abs < 0.001, "supply to room heating temperature difference for #{thermal_zone.name} in the #{building_type}, #{template}, #{climate_zone} model is incorrect. It is #{thermal_zone.sizingZone.zoneHeatingDesignSupplyAirTemperatureDifference}, but should be #{delta_t_r}.")
+          when 'SupplyAirTemperature'
+            setpoint_c = nil
+            tstat = thermal_zone.thermostatSetpointDualSetpoint
+            if tstat.is_initialized
+              tstat = tstat.get
+              setpoint_sch = tstat.heatingSetpointTemperatureSchedule
+              if setpoint_sch.is_initialized
+                setpoint_sch = setpoint_sch.get
+                schedule_types.each do |schedule_type|
+                  full_objtype_name = "OS_Schedule_#{schedule_type}"
+                  if full_objtype_name == setpoint_sch.iddObjectType.valueName.to_s
+                    setpoint_sch = setpoint_sch.public_send("to_Schedule#{schedule_type}").get
+                    # reuse code in Standards.ThermalZone to find tstat max temperature
+                    setpoint_c = std.public_send("schedule_#{schedule_type.downcase}_annual_min_max_value", setpoint_sch)['max']
+                    break
+                  end
+                end
+              end
+            end
+            if setpoint_c.nil?
+              OpenStudio.logFree(OpenStudio::Warn, 'openstudio.Standards.ThermalZone', "#{thermal_zone.name} does not have a valid heating supply air temperature setpoint identified.")
+            else
+              assert(((thermal_zone.sizingZone.zoneHeatingDesignSupplyAirTemperature - setpoint_c).abs - delta_t_r / 9.0 * 5).abs < 0.001, "supply to room heating temperature difference for #{thermal_zone.name} in the #{building_type} #{template}, #{climate_zone} model is incorrect. It is #{(thermal_zone.sizingZone.zoneHeatingDesignSupplyAirTemperature - setpoint_c).abs}, but should be #{delta_t_r}.")
+            end
+          end
+        end
+      else
+        OpenStudio.logFree(OpenStudio::Info, 'openstudio.Standards.ThermalZone', "#{thermal_zone.name} is not a heated zone, skip heating supply air temperature set point difference test.")
+      end
+
+    end
+  end
+
+  # this check uses very similar code to the one that implements this requirement
+  def check_sizing_values(model, building_type, template, climate_zone)
```

Comment:
We probably want to add the YARD comments before merging.

Reply:
done